src/anomalib/metrics/f1_score.py

```diff
@@ -1,36 +1,107 @@
-"""F1 Score metric.
-
-This is added for convenience.
-"""
+"""F1 Score and F1Max Metrics for Binary Classification Tasks."""
 
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-import logging
-from typing import Any, Literal
-
+import torch
+from torchmetrics import Metric
 from torchmetrics.classification import BinaryF1Score
 
-logger = logging.getLogger(__name__)
+from anomalib.metrics.base import AnomalibMetric
+
+from .precision_recall_curve import BinaryPrecisionRecallCurve
+
+
+class F1Score(AnomalibMetric, BinaryF1Score):
+    """Wrapper to add AnomalibMetric functionality to F1Score metric."""
+
+
+class _F1Max(Metric):
```

Comment:
I agree, if they are stored in a single module, it should be named something like `f1.py`?

Reply:
I combined these two metrics in a single module because F1Max is just a special case of the F1Score. Both are implementations of the F1 score. One is computed directly over binary predictions while the other one computes the maximum value that can be achieved using the raw anomaly score predictions.

Reply:
The `prefix` argument is useful in any situation where the user wants to compute the same metric multiple times (in different configurations) in a given test run. The most obvious use case is computing the same metric for image and pixel-level predictions, but this is not the only one.

For example, a user could be interested in computing the F1Score for the same set of predictions at different threshold values, e.g.

```python
f1_40 = F1Score(fields=["pred_score", "gt_label"], threshold=0.4, prefix="0.4_")
f1_50 = F1Score(fields=["pred_score", "gt_label"], threshold=0.5, prefix="0.5_")
f1_60 = F1Score(fields=["pred_score", "gt_label"], threshold=0.6, prefix="0.6_")
SomeModel(metrics=[image_f1_score, pixel_f1_score])
```

This is just a quick example of course, but many other use-cases could exist where the same metric type needs to be computed in different configurations.